src/models/model.cpp

```diff
@@ -36,13 +36,7 @@ static std::wstring CurrentModulePath() {
 
 namespace Generators {
 
-State::State(const GeneratorParams& params) : params_{params.shared_from_this()} {
-  // Add extra user inputs
-  for (auto& input : params.extra_inputs) {
-    input_names_.push_back(input.name.c_str());
-    inputs_.push_back(input.tensor->ort_tensor_.get());
-  }
-}
```

Comment:
Does this change impact other models (whisper) which might depend on extra inputs?
It seems like now every type of Model must have this extra input defined in their class? Can this input be constructed on the parent class instead of constructing them for all the children classes?

Reply:
I now added it for GPT and Whisper, but you're right it would be better if it was constructed in the `State` class. It would require some refactoring though since the `State` class doesn't own a captured graph or doesn't even have a reference on the model, because GPT and Whisper don't support CUDA graph. I'd rather not mess with that flow at the moment, especially since I don't have a good way to teste GPT and Whisper (they don't work on DML).